databricks/sdk/_base_client.py

```diff
@@ -226,6 +242,12 @@ def _perform(self,
                  files=None,
                  data=None,
                  auth: Callable[[requests.PreparedRequest], requests.PreparedRequest] = None):
+        # Keep track of the initial position of the stream so that we can rewind it if
+        # we need to retry the request.
```

Comment:
Can you include an example of where the stream might not be at `0`?

Reply:
Does this need to be a shared variable between all retry attempts (initialized before the first attempt)?

Reply:
This could be needed if users use a single stream to produce a sequence of requests (e.g. upload a large file into chunks). Though, I don't know if we have such use cases at the moment. My real motivation was to be defensive here.

Reply:
> Does this need to be a shared variable between all retry attempts (initialized before the first attempt)?

I don't think it does in this case because we systematically unwind at the end of the call. The benefit of this approach is that it keeps this change pretty minimal. The disadvantage is that we make on too many seek call (see other comment).